functions/that-api-webhooks/hooks/send/activeCampaign.js

```diff
@@ -0,0 +1,245 @@
+  // Sync contact will update or create new contact based on AC's
+  // contact key, email address
+  dlog('call syncContact for %o', contact);
+  const url = `${acBaseUrl}/contact/sync`;
+  const reqConfig = {
+    url,
+    method: 'post',
+    ...axiosReqConfig,
+    data: contact,
+  };
+  return axios(reqConfig)
+    .then(r => {
+      if (![200, 201].includes(r.status)) {
+        dlog('exception syncing contact');
+        Sentry.withScope(scope => {
+          scope.setLevel('error');
+          scope.setContext('exception syncing contact', contact, r);
+          Sentry.captureMessage('exception syncing contact');
+        });
+        return undefined;
+      }
+
+      return r.data.contact;
+    })
+    .catch(err => {
+      dlog('exception syncing contact: %s', err);
+      Sentry.captureException(err);
+      return undefined;
+    });
+}
+
+function searchForTag(tagName) {
+  dlog('search for tag %s', tagName);
+  const url = `${acBaseUrl}/tags`;
+  const reqConfig = {
+    url,
+    method: 'get',
+    ...axiosReqConfig,
+    params: {
+      search: tagName,
+    },
+  };
+  return axios(reqConfig)
+    .then(r => {
+      if (r.status !== 200) {
+        Sentry.withScope(scope => {
+          scope.setLevel('error');
+          scope.setContext('non-200 status searching for tag', { tagName }, r);
+          Sentry.captureMessage('non-200 status searching for tag');
+        });
+        return undefined;
+      }
+      if (r.data && r.data.tags && r.data.tags.length > 1) {
+        dlog(
+          'query for tag, %s, returned %d matches, expect 1 or 0',
+          tagName,
+          r.data.tags.length,
+        );
+        Sentry.withScope(scope => {
+          scope.setLevel('error');
+          scope.setContext(
+            'Tag search return > 1 contact',
+            { tagName },
+            r.data.tags,
+          );
+          Sentry.captureMessage('query by email returned > 1 contact');
+        });
+        return undefined;
+      }
+      return r.data.tags[0];
+    })
+    .catch(err => {
+      dlog('exception searching for tag: %s', err);
+      Sentry.captureException(err);
+      return undefined;
+    });
+}
+
+function addTagToContact(acId, tagId) {
+  dlog('call addTagToContact for id %s adding tag %s', acId, tagId);
+  const url = `${acBaseUrl}/contactTags`;
+  const reqConfig = {
+    url,
+    method: 'post',
+    ...axiosReqConfig,
+    data: {
+      contactTag: {
+        contact: acId,
+        tag: tagId,
+      },
+    },
+  };
+  return axios(reqConfig)
+    .then(r => {
+      if (![200, 201].includes(r.status)) {
+        dlog('issue adding tag to contact');
+        Sentry.withScope(scope => {
+          scope.setLevel('error');
+          scope.setContext(
+            'Issue adding tag to contact',
+            { acId },
+            { tagId },
+            r,
+          );
+          Sentry.captureMessage('issue adding tag to contact');
+        });
+        return undefined;
```

Comment:
this return as undefined combined with the catch's undefined could lead to confusion on the caller.

Reply:
Corrected by adding better error handling